File: packages/firebase_inappmessaging/android/src/main/java/com/example/firebase_inappmessaging/FirebaseInappmessagingPlugin.java
package com.example.firebase_inappmessaging;

import com.google.firebase.inappmessaging.FirebaseInAppMessaging;
import io.flutter.plugin.common.MethodCall;
import io.flutter.plugin.common.MethodChannel;
import io.flutter.plugin.common.MethodChannel.MethodCallHandler;
import io.flutter.plugin.common.MethodChannel.Result;
import io.flutter.plugin.common.PluginRegistry.Registrar;

/** FirebaseInappmessagingPlugin */
public class FirebaseInappmessagingPlugin implements MethodCallHandler {
  private final FirebaseInAppMessaging instance;

  public static void registerWith(Registrar registrar) {

    final MethodChannel channel = new MethodChannel(registrar.messenger(),
          "plugins.flutter.io/firebase_inappmessaging");
    channel.setMethodCallHandler(new FirebaseInappmessagingPlugin());
  }

  private FirebaseInappmessagingPlugin() {
    instance = FirebaseInAppMessaging.getInstance();
  }

  @Override
  public void onMethodCall(MethodCall call, Result result) {
    switch (call.method) {
      case "triggerEvent": {
        String eventName = call.argument("eventName");
        instance.triggerEvent(eventName);
        result.success(null);
        break;
      }
      case "setMessagesSuppressed": {
        boolean suppress = call.argument("suppress");
        instance.setMessagesSuppressed(suppress);
        result.success(null);
        break;
      }
      case "dataCollectionEnabled": {
        boolean dataCollectionEnabled = call.argument("dataCollectionEnabled");
        instance.setAutomaticDataCollectionEnabled(dataCollectionEnabled);
        result.success(null);
        break;
      }
      default: {
        result.notImplemented();
        break;
      }
    }
  }
}

File: packages/firebase_inappmessaging/ios/Classes/FirebaseInappmessagingPlugin.m
#import "FirebaseInappmessagingPlugin.h"

#import <Firebase/Firebase.h>

@implementation FirebaseInappmessagingPlugin
+ (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar>*)registrar {
  FlutterMethodChannel* channel = [FlutterMethodChannel
      methodChannelWithName:@"plugins.flutter.io/firebase_inappmessaging"
            binaryMessenger:[registrar messenger]];
  FirebaseInappmessagingPlugin* instance = [[FirebaseInappmessagingPlugin alloc] init];
  [registrar addMethodCallDelegate:instance channel:channel];
}

- (instancetype)init {
    self = [super init];
    if (self) {
        if (![FIRApp appNamed:@"__FIRAPP_DEFAULT"]) {
            NSLog(@"Configuring the default Firebase app...");
            [FIRApp configure];
            NSLog(@"Configured the default Firebase app %@.", [FIRApp defaultApp].name);
        }
    }
    return self;
}

- (void)handleMethodCall:(FlutterMethodCall*)call result:(FlutterResult)result {
  if ([@"triggerEvent" isEqualToString:call.method]) {
      NSString *eventName = call.arguments[@"eventName"];
      FIRInAppMessaging *fiam = [FIRInAppMessaging inAppMessaging];
      [fiam triggerEvent:eventName];
      result(nil);
  } else if ([@"setMessagesSuppressed" isEqualToString:call.method]) {
      NSNumber *suppress = (NSNumber *) call.arguments[@"suppress"];
      FIRInAppMessaging *fiam = [FIRInAppMessaging inAppMessaging];
      fiam.messageDisplaySuppressed = [suppress boolValue];
      result(nil);
  } else if ([@"dataCollectionEnabled" isEqualToString:call.method]) {
      NSNumber *dataCollectionEnabled = (NSNumber *) call.arguments[@"dataCollectionEnabled"];
      FIRInAppMessaging *fiam = [FIRInAppMessaging inAppMessaging];
      fiam.automaticDataCollectionEnabled = [dataCollectionEnabled boolValue];
      result(nil);
  } else {
    result(FlutterMethodNotImplemented);
  }
}

@end
